core/src/com/unciv/logic/automation/city/ConstructionAutomation.kt

```diff
@@ -24,11 +24,17 @@ class ConstructionAutomation(val cityConstructions: CityConstructions){
     private val cityInfo = cityConstructions.city
     private val civInfo = cityInfo.civInfo
 
-    private val buildings = cityInfo.getRuleset().buildings.values
+    private val buildings = cityInfo.getRuleset().buildings.values.asSequence()
+
     private val nonWonders = buildings.filterNot { it.isAnyWonder() }
+        .filterNot { buildableBuildings[it.name] == false } // if we already know that this building can't be built here then don't even consider it
```

Comment:
Surely buildableBuildings will not be initiated at this point yet?

Reply:
That's a sequence so that predicate would be evaluated after the object is initialized but I went ahead and moved `buildableBuildings` above this statement and did the same with the unit equivalent too.